python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -231,6 +249,79 @@ def run(self, splash=None, version=None):
         result = app.exec_()
         return result
 
+    def uses_legacy_authentication(self):
+        """
+        Returns if the Desktop's bootrap code uses the tk-framework-login for
+        logging in.
+
+        :returns: True the bootstrap logic is older than 1.1.0, False otherwise.
+        """
+        return StrictVersion(self.app_version) < StrictVersion("1.1.0")
+
+    def create_legacy_login_instance(self):
+        """
+        Creates the pre-bootstrap-1.1.0 ShotgunLogin used to authenticate.
+
+        :raises ImportError: Thrown if the module is not available.
+
+        :returns: A tk-framework-login's ShotgunLogin instance.
+        """
+        try:
+            from python import ShotgunLogin
+        except ImportError:
+            self.log_exception("Could not import tk-framework-login")
+            raise
+        else:
+            return ShotgunLogin.get_instance_for_namespace("tk-desktop")
+
+    def _assert_authentication_fault_exists(self, splash):
+        """
+        Asserts that we importing the right version of shotgun_api3. If we are
+        not, the user will be shown a countdown and the app will be restarted.
+        """
+        from tank_vendor import shotgun_api3
```

Comment:
Is there any case you can think of where the user ends up in an infinite loop? What if the internet connection is down and the core auto-update raises an error? Is this a case we should QA?

If I am reading this right, and we are trying to establish if we have a more recent core API available because it was downloaded (but not loaded in memory yet), maybe the logic could be something like this:

```
if the core in memory is pre 0.16:
   if 0.16 is available on disk:
       countdown splash + restart_desktop
   else:
       display error message
```